DownFall/GameViewController.swift

```diff
@@ -127,10 +164,10 @@ extension GameViewController: GameSceneDelegate {
         
     }
     
-    func reset() {
+    func reset(_ scene: SKScene) {
         let fadeOut = SKAction.fadeOut(withDuration: 0.75)
         let remove = SKAction.removeFromParent()
-        gameSceneNode?.run(SKAction.group([fadeOut, remove])) { [weak self] in
+        scene.run(SKAction.group([fadeOut, remove])) { [weak self] in
             self?.startLevel()
         }
     }
```

Comment:
This GameSceneDelegate responsibility has expanded lets rename it. Something to do with Coordinating.

Reply:
It's now called GameSceneCoordinatingDelegate